### ISSUE TITLE
Webshop app should be translatable in Crowdin

### ISSUE THREAD

**Issue by sihma** [frappe/webshop#150]
This also concerns other Frappe project (payments, and most others, except from HRMS)

While ERPNext is [available for translation in Crowdin](https://crowdin.com/project/frappe/fr), the Webshop app strings aren't, which leaves out big translation gaps were if you have, let's say, a website in French with a webshop, the website is translated except for all webshop functionalities.

This brings situation where users provide translations in issues, like in #144.

The app shoud be available for translations and so do the other Frappe apps. What is missing to provide this?

Thanks!

**Comment by clevermind-me** [comment]
Hi! My company is interested in use of Frappe Webshop and we also have an issue about localization of 4 parts of app in Item Review document (other docs will be added later also):

- in the product page “star”, “out of”, month in the review date,
![Снимок экрана 2025-01-07 в 20 40 16](https://github.com/user-attachments/assets/026b0496-d3bb-4179-81d3-90ce5268b900)

- pop-up form “Submit review” in full.
<img width="431" alt="Снимок экрана 2025-01-07 в 20 40 21" src="https://github.com/user-attachments/assets/6503e8c6-92a2-4e3d-bc17-dd32d42e7321" />

I kindly ask you to pay attention to this problem of users, as it is important and gives broad prospects to the application.

**Comment by SmoothPlay** [comment]
Yes, the Item Configuration and Enquiry Form Modals are also not translatable. Anyone found out why yet?
This is one of the most important things to fix regarding translations.

**Comment by clevermind-me** [comment]
Maybe it is because app was not developing since 2022 (you can see stats [here](https://github.com/frappe/webshop/graphs/contributors)) and issues are not being processed.

But this app is more connected to internal modules (stock, selling, CRM and so on), than for example forums or smthng other that can be external and just have connection, so we hope that it will be developed when more resources be free from more important and urgent Frappe tasks or some other reasons, in any later time:)

**Comment by SmoothPlay** [comment]
> Maybe it is because app was not developing since 2022 (you can see stats [here](https://github.com/frappe/webshop/graphs/contributors)) and issues are not being processed.
> 
> But this app is more connected to internal modules (stock, selling, CRM and so on), than for example forums or smthng other that can be external and just have connection, so we hope that it will be developed when more resources be free from more important and urgent Frappe tasks or some other reasons, in any later time:)

That's very unfortunate, the translations of modals is the only thing I need now.

I can't translate any external custom apps for some reason.

Already did the following:

- Moved the file from templates/generators/item  to public/js
- Added the Translation Tags __('Example'),
- Added import './js/item_configure' to web.bundle.js
- Confirmed that web_include_js = "web.bundle.js" is present in hooks.py
- Ran bench build and bench build-message-files
- Also tried bench get-untranslated, but the JS strings and variables are still not there.
- Checked the pt.csv for the translations and they're not there.
- Translations are not working when visiting the page.

Hopefully someone can give at least some insights on why this is happening, I don't mind doing the grunt work.

**Comment by clevermind-me** [comment]
In fact, it is possible to change apps language changing files on server on my system manually.

> the translations of modals is the only thing I need now.

Can you give 1 or 2 examples (place marked on screenshot and write text what is needed). I'll try on my test system and if I can change it, I'll help you with it.

**Comment by SmoothPlay** [comment]
> In fact, it is possible to change apps language changing files on server on my system manually.
> 
> > the translations of modals is the only thing I need now.
> 
> Can you give 1 or 2 examples (place marked on screenshot and write text what is needed). I'll try on my test system and if I can change it, I'll help you with it.

Oh, thank you, I would much apreciate it.

Products Page:
![Image](https://github.com/user-attachments/assets/4e7328d7-c011-43bc-83d6-2592ef9c24d1)
When you are in the Webshop Product Page, and click either Select Variant or Contact Us, it will Open a Modal.

Both those Modals can't be translated. Anything at all.

webshop/templates/generators/item/item_configure.js
![Image](https://github.com/user-attachments/assets/53ef24ca-0799-4383-a333-b61042fb0234)

and

webshop/templates/generators/item/item_inquiry.js
![Image](https://github.com/user-attachments/assets/a83d79aa-d6f6-4f29-b49d-a51c214b4d8f)

Both Modals contain the Translation tags, but simply won't parse them for some reason...

Thank you for your attention, it's nice to feel some support in such lonely struggle :)

PS: HTML translates perfectly. I just don't know how translate those two modals, which seem to be pure Javascript.

**Comment by clevermind-me** [comment]
I don't know how to make these words translatable to different languages. 

But you can change translation to required language inside the file. 
For example I changed this template item_inquiry.js 

And the result is ok if you don't need a multilanguage site

<img width="537" alt="Image" src="https://github.com/user-attachments/assets/a2572be1-288a-4432-827c-65f734d48af1" />

After changing don't forget to use command "bench build" to refresh js files

**Comment by SmoothPlay** [comment]
> I don't know how to make these words translatable to different languages. 
> 
> But you can change translation to required language inside the file. 
> For example I changed this template item_inquiry.js 
> 
> And the result is ok if you don't need a multilanguage site
> 
> <img width="537" alt="Image" src="https://github.com/user-attachments/assets/a2572be1-288a-4432-827c-65f734d48af1" />
> 
> After changing don't forget to use command "bench build" to refresh js files

Thank you so much for your help, that was very kind.Yes, for the inquiry your solution is perfect, we just have to change the labels to html with the translation tags. And it will translate to any language.

For the item_configuration it is much more complicated. As it has dynamic variables.

Just to make sure that it wasn't my installation problem or some bad dependencies, I went and tested the Frappe Cloud version and the problem is there too.

So basically this is something everyone is experiencing and it seems like nobody has a way to fix it so far or even care about it.

I guess I'm placing the cart before the horse.

I'm going to try to get around it in some hard way and see how it goes.

Thank you so much again, if you happen to need anything feel free to ask, I've been tweaking Frappe more than 2 years and I'm very close to being finally stable and happy with it.

**Comment by clevermind-me** [comment]
> For the item_configuration it is much more complicated. As it has dynamic variables.

Maybe I don't completely understand what you really need, but I used same method and it works for variant selection

<img width="539" alt="Image" src="https://github.com/user-attachments/assets/8c835a9d-06cf-4696-89fa-c3e4bb4fd894" />

<img width="1258" alt="Image" src="https://github.com/user-attachments/assets/389430bd-3231-4146-9ec7-02fdddc404a4" />

Can you send me link to your website?

> Thank you so much again, if you happen to need anything feel free to ask, I've been tweaking Frappe more than 2 years and I'm very close to being finally stable and happy with it.

Ok, thank you

**Comment by SmoothPlay** [comment]
> Maybe I don't completely understand what you really need, but I used same method and it works for variant selection

Wow, That's exactly what I want to translate. The Attributes. Although I'm aiming for a multi-language site I'm still very curious on your method to translate the Attribute. Or did you give the name of the Attribute in the same language?

I'm working on a way to translate without the need of Build every time. 

> <img alt="Image" width="539" src="https://private-user-images.githubusercontent.com/181402779/433512988-8c835a9d-06cf-4696-89fa-c3e4bb4fd894.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ3MDQ5MzgsIm5iZiI6MTc0NDcwNDYzOCwicGF0aCI6Ii8xODE0MDI3NzkvNDMzNTEyOTg4LThjODM1YTlkLTA2Y2YtNDY5Ni04OWZhLWMzZTRiYjRmZDg5NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxNVQwODEwMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZDYzMWY1ZjY1YmNhMWQ5ZmM4YWMyMDVmNzM4ZjlhNjM3MGJlZGJlOTU4ZGUxZmRjNDU2OTI0OGQ3NTFjMTA2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.En7t_cPyUl-k4lBX4XYLss4kL2UnoDDDXI-i4EgzA24"> 

> Can you send me link to your website?

I would prefer to do that In private if possible. I'll send you a message.

**Comment by clevermind-me** [comment]
> > Maybe I don't completely understand what you really need, but I used same method and it works for variant selection
> 
> Wow, That's exactly what I want to translate. The Attributes. Although I'm aiming for a multi-language site I'm still very curious on your method to translate the Attribute. Or did you give the name of the Attribute in the same language?

Yes, names of Attributes I gave in my language. I don't know how to make Attributes translatable.

> > Can you send me link to your website?
> 
> I would prefer to do that In private if possible. I'll send you a message.

I added email-address in my profile. It would be interesting to see what you already realized.
